### PR TITLE
EPMDJ-7965 TestNG doesn't track skipped test

### PR DESCRIPTION
**Pull request by Mazel-Tovr**
also add Test metadata

### CODE REVIEW

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/testng/TestNGStrategy.kt [hunk 1]
```diff
@@ -145,7 +158,35 @@ object TestNGStrategy : AbstractTestStrategy() {
...
     }
+
+    /**
+     * Support for tests, disabled by changing flag in @Test(enabled = false) annotation
+     * When engine version is lower then 7.4.0 also support @Ignore annotation
+     */
+    private fun CtClass.disabledTestsSupport() = getDeclaredMethod("run").insertAfter(
+            """
+                java.util.Iterator disabledTests = getExcludedMethods().iterator();
+                while(disabledTests.hasNext()){
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/testng/TestNGStrategy.kt:171]: formatting

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/testng/TestNGStrategy.kt [hunk 2]
```diff
@@ -145,7 +158,35 @@ object TestNGStrategy : AbstractTestStrategy() {
...
+                    $packagePath.ITestNGMethod test = ($packagePath.ITestNGMethod) disabledTests.next();
+                    ${TestListener::class.java.name}.INSTANCE.${TestListener::testIgnored.name}("${engineSegment}/[class:" + test.getTestClass().getName() + "]/[method:" + test.getMethodName() + "]");
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/testng/TestNGStrategy.kt:173]:
```suggestion
                    ${TestListener::class.java.name}.INSTANCE.${TestListener::testIgnored.name}("$engineSegment/[class:" + test.getTestClass().getName() + "]/[method:" + test.getMethodName() + "]");     
```

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/testng/TestNGStrategy.kt [hunk 3]
```diff
@@ -145,7 +158,35 @@ object TestNGStrategy : AbstractTestStrategy() {
...
+    /**
+     * Support 7.4.0 testng @Ignore annotation
+     */
+    private fun CtClass.ignoredTestSupport() = getMethod(
+        "isAnnotationPresent",
+        "(Lorg/testng/internal/annotations/IAnnotationFinder;Ljava/lang/reflect/Method;Ljava/lang/Class;)Z"
+    ).insertAfter(
+        """ 
+            if ($3 == $packagePath.annotations.IIgnoreAnnotation.class && ${'$'}_) {
+                ${TestListener::class.java.name}.INSTANCE.${TestListener::testIgnored.name}("${engineSegment}/[class:" + $2.getDeclaringClass().getName() + "]/[method:" + $2.getName() + "]");
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/testng/TestNGStrategy.kt:187]:
```suggestion
                ${TestListener::class.java.name}.INSTANCE.${TestListener::testIgnored.name}("$engineSegment/[class:" + $2.getDeclaringClass().getName() + "]/[method:" + $2.getName() + "]");
```

File: test/cucumber/cucumber-v4/src/test/java/com/automatedtest/sample/test/TestSteps.java [hunk 4]
```diff
@@ -19,6 +19,7 @@
 import com.epam.drill.plugins.test2code.api.*;
 import com.epam.drill.test.agent.instrumentation.testing.cucumber.*;
 import com.epam.drill.test.common.*;
+import com.epam.drill.test.common.TestData;
```
Comment (LikhanovMaxim) [inline test/cucumber/cucumber-v4/src/test/java/com/automatedtest/sample/test/TestSteps.java:22]: why do you add?

**Review by LikhanovMaxim — CHANGES_REQUESTED** [review]

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/testng/TestNGStrategy.kt [hunk 5]
```diff
@@ -41,8 +45,17 @@ object TestNGStrategy : AbstractTestStrategy() {
         classLoader: ClassLoader?,
         protectionDomain: ProtectionDomain?,
     ): ByteArray? {
-        createTestListener(pool, classLoader, protectionDomain)
-        ctClass.constructors.forEach { it.insertAfter("addTestListener(new $drillListener());") }
+        if (ctClass.isTestRunner()) {
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/testng/TestNGStrategy.kt:48]: create techDebt

**Review by LikhanovMaxim — CHANGES_REQUESTED** [review]

File: test/cucumber/cucumber-v4/src/test/java/com/automatedtest/sample/test/TestSteps.java (hunk 4 above)
Comment (Mazel-Tovr) [reply test/cucumber/cucumber-v4/src/test/java/com/automatedtest/sample/test/TestSteps.java:22]: In the new version of the t2 api, a class with exactly the same name appeared, so we had to explicitly import this class

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/Strategy.kt [hunk 6]
```diff
@@ -20,6 +20,7 @@ import javassist.ClassPool
 import javassist.CtClass
 import java.security.ProtectionDomain
 
+//TODO EPMDJ-83109 Rework the choice of strategies
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/Strategy.kt:23]:
```suggestion
//TODO EPMDJ-8319 Rework the choice of strategies
```

**Review by LikhanovMaxim — APPROVED** [review]

**Pull request merged**